client/app/visualizations/table/columns/datetime.js

```diff
@@ -1,6 +1,45 @@
-/* eslint-disable react/prop-types */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Input from 'antd/lib/input';
+import Popover from 'antd/lib/popover';
+import Icon from 'antd/lib/icon';
 import { createDateTimeFormatter } from '@/lib/value-format';
 
+function Editor({ column, onChange }) {
+  return (
+    <React.Fragment>
+      <div className="m-b-15">
+        <label htmlFor={`table-column-editor-${column.name}-datetime-format`}>
+          Date/Time format
+          <Popover
+            content={(
+              <React.Fragment>
+                Format&nbsp;
+                <a href="https://momentjs.com/docs/#/displaying/format/" target="_blank" rel="noopener noreferrer">specs.</a>
+              </React.Fragment>
+            )}
+          >
+            <Icon className="m-l-5" type="question-circle" theme="filled" />
+          </Popover>
+        </label>
+        <Input
+          id={`table-column-editor-${column.name}-datetime-format`}
+          defaultValue={column.dateTimeFormat}
+          onChange={event => onChange({ dateTimeFormat: event.target.value })}
+        />
+      </div>
+    </React.Fragment>
+  );
+}
+
+Editor.propTypes = {
+  column: PropTypes.shape({
+    name: PropTypes.string.isRequired,
+    dateTimeFormat: PropTypes.string,
+  }).isRequired,
+  onChange: PropTypes.func.isRequired,
+};
```

Comment:
Couldn't figure out why but if I create a new query `select 1556668800000 as date` and then edit the column to be "date/time" it won't react to a format input. I enter "MMM-YYYY" but the viz preview stays the number :/

Reply:
Cast it to date/time type so redash will convert it to `moment` instance. it's quite weird, but changing this behavior is not related to this PR

Reply:
Ohhh yeah that's weird  😒

Reply:
It's definitely confusing and not how the other types work -- let's open a separate issue for this.

Reply:
PR getredash/redash#4389